From 9d0bb3bf6590919529cf67609da293766cb00843 Mon Sep 17 00:00:00 2001
From: W Anders <w@doubleu.codes>
Date: Fri, 1 Jan 2021 07:51:54 -0600
Subject: [PATCH 06/13] Backport: Add CentOS Stream to xe-linux-distribution

"CentOS Stream" is specified in /etc/centos-release, along with only the
major version. Updates the expression in the identify_redhat function to
include a check for this distribution and prevent daemon errors.
---
 mk/xe-linux-distribution | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/mk/xe-linux-distribution b/mk/xe-linux-distribution
index ed628a9..af57bfb 100755
--- a/mk/xe-linux-distribution
+++ b/mk/xe-linux-distribution
@@ -123,6 +123,7 @@ identify_redhat()
     # CentOS release 4.0 (Final)
     # CentOS release 5 (Final)
     # CentOS Linux release 7.0.1406 (Core)
+    # CentOS Stream release 8
 
     # distro=cloudlinux
     # CloudLinux Server release 6.10 (Vladimir Lyakhov)
@@ -153,6 +154,7 @@ identify_redhat()
                -e 's/^CentOS release \([0-9]*\)\.\([0-9]*\) (.*)/distro=centos;major=\1;minor=\2/gp;' \
                -e 's/^CentOS release \([0-9]*\) (.*)/distro=centos;major=\1/gp;' \
                -e 's/^CentOS Linux release \([0-9]*\)\.\([0-9]*\).*$/distro=centos;major=\1;minor=\2/gp;' \
+               -e 's/^CentOS Stream release \([0-9]*\).*$/distro=centos;major=\1/gp;' \
                -e 's/^CloudLinux\( Linux|Server\)\? release \([0-9]*\)\.\([0-9]*\)\(\.[0-9]*\)\? (.*)/distro=cloudlinux;major=\2;minor=\1/gp;' \
                -e 's/^CloudLinux release \([0-9]*\)\.\([0-9]*\)\(\.[0-9]*\)\? (.*)/distro=cloudlinux;major=\1;minor=\2/gp;' \
                -e 's/^Enterprise Linux Enterprise Linux .* release \([0-9]*\)\.\([0-9]*\) (.*)$/distro=oracle;major=\1;minor=\2;/gp;' \
-- 
2.30.2

